source/behaviors/offClick/offClick.ts

```diff
@@ -29,16 +30,36 @@ export class OffClickDirective implements OnInit, OnDestroy {
 	}
 
 	ngOnInit() {
-		setTimeout(() => {
-			document.addEventListener('click', this.listener);
-		}, 0);
+		if (this.active) {
+			setTimeout(() => {
+				this.addListener();
+			});
+		}
 	}
 
-	ngOnDestroy() {
+	ngOnChanges(changes: SimpleChanges): void {
```

Comment:
That'll work I guess. I also realized I forgot to clean up this implementation anyway, to get rid of the string typing. Usually when we do use ngOnChanges we create an interface so we can get strong typing on the changes object.